test/unit/test_diskfile.py

```diff
@@ -15,234 +15,416 @@
-    resp.reason = "Sproxyd Internal Failure"
-    return resp
+    def __init__(self, resp_status=200):
+        self.resp_status = resp_status
+        self._buffer = StringIO.StringIO()
 
+    def getresponse(self):
+        return FakeHTTPResp(self.resp_status)
 
-def _mock_conn_getresponse_timeout(self, conn):
-    """Simulate an error 500."""
-    resp = FakeHTTPResponse()
-    resp.status = 200
-    eventlet.sleep(PROXY_TIMEOUT + 0.1)
-    return resp
-
-
-class FakeConn:
-    """fake response class for faking HTTP answers."""
+    def send(self, data):
+        self._buffer.write(data)
 
     def close(self):
-        """Close conn."""
         pass
 
 
-def _mock_do_connect(self, ipaddr, port, method, path, headers=None,
-                     query_string=None, ssl=False):
-    conn = FakeConn()
-    return conn
-
-
-class TestSproxydDiskFile(unittest.TestCase):
-    """Tests for swift_scality_backend.diskfile."""
-
-    def setUp(self):
-        self._orig_tpool_exc = eventlet.tpool.execute
-        eventlet.tpool.execute = lambda f, *args, **kwargs: f(*args, **kwargs)
-        self.fake_logger = logging.getLogger(__name__)
+class TestSproxydFileSystem(unittest.TestCase):
+    """Tests for swift_scality_backend.diskfile.SproxydFileSystem"""
+
+    def test_init_with_default_timeout_values(self):
+        sfs = SproxydFileSystem({}, mock.Mock())
+        self.assertEqual(10, sfs.conn_timeout)
+        self.assertEqual(3, sfs.proxy_timeout)
+
+    def test_init_with_custom_timeout_values(self):
+        conf = {'sproxyd_conn_timeout': 42.1, 'sproxyd_proxy_timeout': 4242.1}
+        sfs = SproxydFileSystem(conf, mock.Mock())
+        self.assertEqual(42.1, sfs.conn_timeout)
+        self.assertEqual(4242.1, sfs.proxy_timeout)
+
+    def test_init_with_default_splice(self):
+        sfs = SproxydFileSystem({}, mock.Mock())
+        self.assertFalse(sfs.use_splice)
+
+    def test_init_with_splice_no(self):
+        sfs = SproxydFileSystem({'splice': 'no'}, mock.Mock())
+        self.assertFalse(sfs.use_splice)
+
+    def test_init_base_path_has_slashes(self):
+        conf = {'sproxyd_path': 'missing_slashes'}
+        sfs = SproxydFileSystem(conf, mock.Mock())
+        self.assertEqual('/missing_slashes/', sfs.base_path)
+
+    def test_init_sproxyd_hosts(self):
+        # Mind the white spaces
+        conf = {'sproxyd_host': ' host1:81 , host2:82 '}
+        sfs = SproxydFileSystem(conf, mock.Mock())
+        expected_sproxyd_hosts_set = set([('host1', 81), ('host2', 82)])
+        self.assertEqual(expected_sproxyd_hosts_set, sfs.sproxyd_hosts_set)
+
+    def test_init_monitoring_threads(self):
+        conf = {'sproxyd_host': 'host1:81,host2:82'}
+        sfs = SproxydFileSystem(conf, mock.Mock())
+        self.assertEqual(2, len(sfs.healthcheck_threads))
+
+    def _test_init_splice_unavailable(self):
+        sfs = SproxydFileSystem({'splice': 'no'}, mock.Mock())
+        self.assertFalse(sfs.use_splice)
```

Comment:
There are multiple assertions in this (and subsequent) tests. Consider adding a message.